torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -0,0 +1,231 @@
+# mypy: allow-untyped-decorators
+# mypy: allow-untyped-defs
+
+
+import torch
+import torch.utils._pytree as pytree
+from torch._C import DispatchKey
+from torch._dispatch.python import suspend_functionalization
+from torch._higher_order_ops.utils import (
+    _from_fun,
+    _maybe_reenter_make_fx,
+    clone_outputs_aliasing_inputs,
+    get_dummy_aot_autograd_config,
+    prepare_fw_with_masks,
+    reenter_make_fx,
+)
+from torch._ops import HigherOrderOperator
+from torch._subclasses import FakeTensorMode
+from torch._subclasses.functional_tensor import disable_functional_mode
+from torch.fx.experimental.proxy_tensor import (
+    disable_proxy_modes_tracing,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+)
+from torch.fx.graph_module import GraphModule
+
+
+invoke_subgraph_counter = 0
+
+
+class InvokeSubgraphHOP(HigherOrderOperator):
+    def __init__(self) -> None:
+        super().__init__("invoke_subgraph")
+
+    # identifier is setup by upper part of the stack. This helps us in
+    # identifying two invoke_subgraph calls have same subgraph.
+    def __call__(
+        self,
+        subgraph: GraphModule,
+        identifier: str,
+        operands,
+    ):
+        return super().__call__(subgraph, identifier, operands)
+
+
+invoke_subgraph = InvokeSubgraphHOP()
+
+
+def create_invoke_subgraph_op(subgraph, operands):
+    # Helper to set the identifier for the invoke_subgraph op. We ensure that
+    # each identifier is different here. Its torch.compile responsibility to
+    # replace identifiers for subgraphs that are identical.
+    global invoke_subgraph_counter
+    identifier = f"invoke_subgraph_{invoke_subgraph_counter}"
+    invoke_subgraph_counter += 1
+    return invoke_subgraph(subgraph, identifier, operands)
```

Comment:
Why do we need the identifier at all here? Can we just set it to None? Then torch.compile's responsibility is to figure out how to dedup invoke_subgraph whose identifiers are None

Reply:
The benefit of this is that we wouldn't need this function anymore, instead we'd just be running tests on invoke_subgraph